Sources/Core/JavaModelRenderer.swift

```diff
@@ -222,10 +222,25 @@ public struct JavaModelRenderer: JavaFileRenderer {
     }
 
     func renderBuilderBuild() -> JavaIR.Method {
-        let params = (transitiveProperties.map { param, _ in
+        let constructorParams = (transitiveProperties.map { param, _ in
             "this." + Languages.java.snakeCaseToPropertyName(param)
         } + ["this._bits"]).joined(separator: ",\n")
-        return JavaIR.method(annotations: [.nonnull], [.public], "\(className) build()") { ["return new " + self.className + "(", params, ");"] }
+
+        let missingFieldsCheck = !transitiveProperties.filter { $1.nullability == Nullability.nonnull }.isEmpty
+            ? ["List<String> missingFields = new ArrayList<>();"] + transitiveProperties.filter {
+                $1.nullability == Nullability.nonnull
+            }.map { param, _ in
+                JavaIR.ifBlock(condition: "this." + Languages.java.snakeCaseToPropertyName(param) + " == null") { [
+                    "missingFields.add(\"" + Languages.java.snakeCaseToPropertyName(param) + "\");",
+                ] }
+            } + [JavaIR.ifBlock(condition: "missingFields.size() > 0") { [
```

Comment:
Is usually prefer `!.isEmpty()` to `.size() > 0` checks.

Reply:
Done